roofit/roofit/src/RooLagrangianMorphFunc.cxx

```diff
@@ -0,0 +1,3522 @@
+
+inline bool begins_with(const std::string &input, const std::string &match) {
+  return input.size() >= match.size() &&
+         equal(match.begin(), match.end(), input.begin());
+}
+
+///////////////////////////////////////////////////////////////////////////////
+/// (-?-)
+
+inline TString makeValidName(const char *input) {
+  TString retval(input);
+  retval.ReplaceAll("/", "_");
+  retval.ReplaceAll("^", "");
+  retval.ReplaceAll("*", "X");
+  retval.ReplaceAll("[", "");
+  retval.ReplaceAll("]", "");
+  return retval;
+}
+
+//////////////////////////////////////////////////////////////////////////////
+/// concatenate the names of objects in a collection to a single string
+
+template <class List> std::string concatNames(const List &c, const char *sep) {
+  std::stringstream ss;
+  bool first = true;
+  for (auto itr : c) {
+    if (!first)
+      ss << sep;
+    ss << itr->GetName();
+    first = false;
+  }
+  return ss.str();
+}
+
+///////////////////////////////////////////////////////////////////////////////
+/// this is a workaround for the missing implicit conversion from
+/// SuperFloat<>double
+
+template <class A, class B> inline void assignElement(A &a, const B &b) {
+  a = static_cast<A>(b);
+}
+///////////////////////////////////////////////////////////////////////////////
+// read a matrix from a stream
+
+template <class MatrixT>
+inline MatrixT readMatrixFromStreamT(std::istream &stream) {
+  std::vector<std::vector<SuperFloat>> matrix;
+  std::vector<SuperFloat> line;
+  while (!stream.eof()) {
+    if (stream.peek() == '\n') {
+      stream.get();
+      stream.peek();
+      continue;
+    }
+    SuperFloat val;
+    stream >> val;
+    line.push_back(val);
+    while (stream.peek() == ' ' || stream.peek() == '\t') {
+      stream.get();
+    }
+    if (stream.peek() == '\n') {
+      matrix.push_back(line);
+      line.clear();
+    }
+  }
+  MatrixT retval(matrix.size(), matrix.size());
+  for (size_t i = 0; i < matrix.size(); ++i) {
+    if (matrix[i].size() != matrix.size()) {
+      std::cerr << "matrix read from stream doesn't seem to be square!"
+                << std::endl;
+    }
+    for (size_t j = 0; j < matrix[i].size(); ++j) {
+      assignElement(retval(i, j), matrix[i][j]);
+    }
+  }
+  return retval;
+}
+
+///////////////////////////////////////////////////////////////////////////////
+/// read a matrix from a text file
+
+template <class MatrixT> inline MatrixT readMatrixFromFileT(const char *fname) {
+  std::ifstream in(fname);
+  if (!in.good()) {
+    std::cerr << "unable to read file '" << fname << "'!" << std::endl;
+  }
+  MatrixT mat = readMatrixFromStreamT<MatrixT>(in);
+  in.close();
+  return mat;
+}
+
+///////////////////////////////////////////////////////////////////////////////
+/// convert a TH1* param hist into the corresponding ParamSet object
+
+template <class T> inline std::map<const std::string, T> readValues(TH1 *h_pc) {
+  std::map<const std::string, T> point;
```

Comment:
```suggestion
template <class T>
void readValues(std::map<const std::string, T> & myMap, TH1 *h_pc) {
```
Let's try to always fill these parameter maps in place without copying. Otherwise we'll get more clang errors on macOS.